analyzer/consensus/consensus.go

```diff
@@ -275,20 +275,21 @@ func (m *Main) processBlock(ctx context.Context, height int64) error {
 
 	// Prepare and perform updates.
 	batch := &storage.QueryBatch{}
+	queries := make([]storage.QueryBatch, 5)
```

Comment:
Instead of hardcoding the `5`, you can probably extend the array as you go:
```go
	queries := make([]storage.QueryBatch, 0)
...
		func(f prepareFunc) {
			batch := storage.QueryBatch{}
			queries = append(queries, &batch)
			group.Go(func() error {
				return f(groupCtx, height, &batch)
			})
		}(f)
```

Go has non-straightforward rules around references, loops, and goroutines, but I believe this is fine -- `batch` is created locally inside the loop, so each iteration has its own `batch`. As long as the garbage collector doesn't free it prematurely.

Reply:
`append` may reallocate the array that backs `queries` and we'll lose writes

Reply:
Hm, `queries` only contains _pointers_ to batches (I see I specified the type incorrectly in `make` above; fixed now). So reallocations should move the pointers around but leave their values and the batches untouched, right?

Reply:
o ok

Reply:
Updated! Thanks for the discussion